service/resource/vaultcrt/patch.go

```diff
@@ -0,0 +1,38 @@
+package vaultcrt
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
+	create, err := r.newCreateChange(ctx, obj, currentState, desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	update, err := r.newUpdateChange(ctx, obj, currentState, desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	patch := framework.NewPatch()
+	patch.SetCreateChange(create)
+	patch.SetUpdateChange(update)
+
+	return patch, nil
+}
+
+func (r *Resource) NewDeletePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
```

Comment:
I would like to keep these methods in the update and delete files. Adding a new patch file does not feel like the right thing to me.

Reply:
They aren't in 1:1 relation. Any `New*Patch` may use any `new*Change`. I wouldn't feel comfortable with that.

Reply:
It is not about where they are used but what they do. The creation magic belongs to the create file. The update magic belongs to the update file. The deletion magic belongs to the delete file. It was like this and I want to keep it like this.

Reply:
Where NewUpdatePatch belongs to when it references create and update magic?

Reply:
The result is the update patch. So it belongs to the update file IMO.

Reply:
I think those are separate things. `create changes` may be different for `update patch` and `delete patch`. Both update & delete patches may contain all types of changes. To name a few cases. I treat `patch.go` as a dispatcher of changes which has to be applied. I can disagree and commit though.

Reply:
When they are different there should be something like `newCreateChangeForDeletePatch`. This should then be within the create file, because it is about the create state. 

> I don't like asymmetry in create.go file.

For the the asymmetry is there with the patch file. It is just a new file that makes everything different and I cannot see the clear line anymore. I think the rule is using the file defined by `*` when having the scheme `New*Patch` or `Apply*Change`. That is a pretty good guide to decide where to put code.

Reply:
I don't like it. But I disagree and commit to make this going.
Done.